src/community/leading_eigenvector.c

```diff
@@ -1005,31 +1006,33 @@ int igraph_le_community_to_membership(const igraph_matrix_t *merges,
     igraph_vector_t fake_memb;
     long int components, i;
 
-    if (igraph_matrix_nrow(merges) < steps) {
-        IGRAPH_ERROR("`steps' to big or `merges' matrix too short", IGRAPH_EINVAL);
+    if (no_of_nodes > 0) {
+            components = (long int) igraph_vector_max(membership) + 1;
+    } else {
+        components = 0;
     }
-
-    components = (long int) igraph_vector_max(membership) + 1;
     if (components > no_of_nodes) {
-        IGRAPH_ERROR("Invalid membership vector, too many components", IGRAPH_EINVAL);
+        IGRAPH_ERRORF("Invalid membership vector, number of components is %ld, but should "
+         "not be greather than number of nodes, which is %ld.", IGRAPH_EINVAL, components, no_of_nodes);
     }
     if (steps >= components) {
-        IGRAPH_ERROR("Cannot make `steps' steps from supplied membership vector",
-                     IGRAPH_EINVAL);
+        IGRAPH_ERRORF("Number of steps should be smaller than number of components. "
+                     "Found %" IGRAPH_PRId " steps, %ld components.",
+                     IGRAPH_EINVAL, steps, components);
     }
 
     IGRAPH_VECTOR_INIT_FINALLY(&fake_memb, components);
 
     /* Check membership vector */
     for (i = 0; i < no_of_nodes; i++) {
         if (VECTOR(*membership)[i] < 0) {
-            IGRAPH_ERROR("Invalid membership vector, negative id", IGRAPH_EINVAL);
+            IGRAPH_ERRORF("Invalid membership vector, negative id found: %g.", IGRAPH_EINVAL, VECTOR(*membership)[i]);
         }
         VECTOR(fake_memb)[ (long int) VECTOR(*membership)[i] ] += 1;
     }
     for (i = 0; i < components; i++) {
         if (VECTOR(fake_memb)[i] == 0) {
-            IGRAPH_ERROR("Invalid membership vector, empty cluster", IGRAPH_EINVAL);
+            IGRAPH_ERRORF("Invalid membership vector, empty cluster found, with index %ld.", IGRAPH_EINVAL, i);
```

Comment:
One concern I have is that some of the igraph interfaces use 0-based indexing (Python, C) and others use 1-based indexing (R, Mathematica). This is useful—they match the host language.

But if we report a zero-based index in an error message, it would be confusing to R users.

This didn't really come up before as we didn't report specific values in error messages. Of course, being specific is generally good ...

@ntamas @vtraag Any suggestions about how to proceed here?

 1. Not report indices at all (not ideal—dimishes usefuless)
 2. Spell out that this is a zero-based index (pretty ugly, but at least useful ... )

Reply:
I would go for 1 for sake of simplicity; we were quite okay so far without spelling these out in error messages.

Reply:
I'm a bit torn between the two. 

It might make debugging some errors easy if we immediately get an error reported that reflects the index. However, it might indeed confuse some users, even if we explicitly state that the index is zero-based. Still, it might also save some users some headache if they have an idea where a particular problem may occur. All considering, I think that I would have a (slight) preference of option (2) over option (1).

So @szhorvat, you can break the tie in the decision :)

Reply:
That would indeed be horribly confusing. I'm not sure how much we want to limit error lengths.
We could do `"...index %ld (for 0-based indexing) / %ld (for 1-based indexing)", i, i+1` .
Which can still be confusing for users who are not familiar with this, but at least they can look it up.

Reply:
We actually also already have an ugly `USING_R` in place, so we could also add a conversion function `igraph_index(long idx)` that either simply returns `idx` or `idx + 1` is `USING_R` is defined. Something similar could potentially be done for Mathematica. The downside of this is that is renders unclear for a particular error whether it was thrown from an R / Mathematica session or from a C / Python session, so it may actually only complicate things.

Perhaps it would be simplest to only report the zero-based index, and explicitly mention that.